syntax = "proto3";

package account;

import "google/protobuf/empty.proto";

service UserBaseController {
    rpc List(UserBaseListRequest) returns (stream UserBase) {}
    rpc Create(UserBase) returns (UserBase) {}
    rpc Retrieve(UserBaseRetrieveRequest) returns (UserBase) {}
    rpc Update(UserBase) returns (UserBase) {}
    rpc Destroy(UserBase) returns (google.protobuf.Empty) {}
}

message UserBase {
    string email = 1;
    string user_name = 2;
    string name = 3;
}

message UserBaseListRequest {
}

message UserBaseRetrieveRequest {
    int64 id = 1;
}
